com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoopDef.hlsl

```diff
@@ -121,6 +121,26 @@ float4 SampleEnv(LightLoopContext lightLoopContext, int index, float3 texCoord,
             float3 capturedForwardWS = _Env2DCaptureForward[index].xyz;
             if (dot(capturedForwardWS, texCoord) < 0.0)
                 color.a = 0.0;
+            else
+            {
+                // Controls the blending on the edges of the screen
+                const float amplitude = 100.0;
+
+                float2 rcoords = abs(saturate(ndc.xy) * 2.0 - 1.0);
+
+                // When the object normal is not aligned with the reflection plane, the reflected ray might deviate too much and go out
+                // of the reflection frustum. So we apply blending when the reflection sample coords are on the edges of the texture
+                // These "edges" depend on the screen space coordinates of the pixel, because it is expected that a pixel on the
+                // edge of the screen will sample on the edge of the texture
+
+                // Blending factors taking the above into account
+                bool2 blend = (positionNDC < ndc.xy) ^ (ndc.xy < 0.5);
+                float2 alphas = saturate(amplitude * abs(ndc.xy - positionNDC));
+                alphas = float2(Smoothstep01(alphas.x), Smoothstep01(alphas.y));
+
+                float2 weights = lerp(1.0, saturate(2.0 - 2.0 * rcoords), blend * alphas);
```

Comment:
let's discuss why the weight calculation is here

Reply:
Because I need the ndc coords of the pixel in the camera space of the reflection (the `ndc` variable in the function), so doing the calculation outside of the function requires to get the matrix and do the multiplication again